tests/unit/services/chat-test.js

```diff
@@ -0,0 +1,114 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import MockSocketService from '../../mocks/services/socket';
+
+module('Unit | Service | chat', function (hooks) {
+  setupTest(hooks);
+
+  const setup = testContext => {
+    testContext.owner.register('service:socket', MockSocketService);
+    let chatService = testContext.owner.lookup('service:chat');
+    let lobbyChannel = chatService.socket.socket.channels["rooms:lobby"];
+
+    return {
+      chatService, lobbyChannel
+    }
+  }
+
+  test('it receives messages', function (assert) {
+    let { chatService, lobbyChannel } = setup(this);
+
+    for (let i = 0; i < 3; i++) {
+      // dispatches the callback attached to first arg
+      // the second arg is passed to the callback fn
+      lobbyChannel.dispatch("new:msg", "test message number " + (i + 1))
+    }
+
+    assert.ok(chatService.messages.content.includes("test message number 1"));
+    assert.ok(chatService.messages.content.includes("test message number 2"));
+    assert.ok(chatService.messages.content.includes("test message number 3"));
+    assert.notOk(chatService.messages.content.includes("This message was never sent"));
+  });
+
+  test('it lets you log in if you\'re cool', function (assert) {
+     let { chatService, lobbyChannel } = setup(this);
+
+    lobbyChannel.dispatch("authorized", { user: "daniel \"cool guy\" fridkin" });
+
+    assert.equal(chatService.username, "daniel \"cool guy\" fridkin");
+    assert.ok(chatService.joinedChat);
+
+  });
+
+  test('it doesn\'t let you log in if you\'re not cool', function (assert) {
+     let { chatService, lobbyChannel } = setup(this);
+
+    // Normally, this would result in a browser alert.
+    // But testem literally does not allow you to do that.
+    // So instead MockChannel calls the "new:msg" callback instead
+    lobbyChannel.dispatch("notauthorized", { error: "you suck" });
+    assert.ok(chatService.messages.content.includes("you suck"));
+
+  });
+
+  test('it notifies you when someone leaves chat', function (assert) {
+
+     let { chatService, lobbyChannel } = setup(this);
+
+    lobbyChannel.dispatch("user:left", { user: "lame_nerd23", timestamp: "before the party started" });
```

Comment:
Could we change this to just "nerd23" here? Lame can be interpreted as ableist. Been guilty of it myself many times.
https://github.com/emberjs/guides/pull/1749

Reply:
Oh absolutely